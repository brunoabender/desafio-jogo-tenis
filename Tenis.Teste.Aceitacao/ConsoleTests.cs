using ApprovalTests.Reporters;
using ApprovalTests;
using Xunit;

namespace Tenis.Teste.Aceitacao
{

    [Collection("RecursoCompartilhado")]
    public class ConsoleTests
    {

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public void DevePontuarParaCadaJogadorEFinalizar()
        {
            // Arrange
            var writer = new StringWriter();
            Console.SetOut(writer);

            // Simular a entrada do usuário: 1 para pontuar o primeiro jogador, 2 para pontuar o segundo jogador, "exit" para sair
            var input = new StringReader("1\n2\nsaida\n");
            Console.SetIn(input);

            // Act
            Program.Main();

            // Assert
            var output = writer.ToString();
            Approvals.Verify(output);
        }

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public void DeveMostrarDeuceEFinalizar()
        {
            // Arrange
            var writer = new StringWriter();
            Console.SetOut(writer);

            var input = new StringReader("1\n1\n1\n2\n2\n2\nsaida\n");
            Console.SetIn(input);

            // Act
            Program.Main();

            // Assert
            var output = writer.ToString();
            Approvals.Verify(output);
        }
    }
}
