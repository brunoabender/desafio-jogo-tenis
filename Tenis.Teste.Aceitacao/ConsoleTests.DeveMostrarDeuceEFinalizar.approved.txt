Placar de Tênis:
Jogador 1: 0 sets, 0 games, 0 pontos no game atual
Jogador 2: 0 sets, 0 games, 0 pontos no game atual
Próximo saque: Primeiro Jogador
Modo: Normal
Opções:
Pontuar Jogador 1: 1
Pontuar Jogador 2: 2
Placar de Tênis:
Jogador 1: 0 sets, 0 games, 15 pontos no game atual
Jogador 2: 0 sets, 0 games, 0 pontos no game atual
Próximo saque: Primeiro Jogador
Modo: Normal
Opções:
Pontuar Jogador 1: 1
Pontuar Jogador 2: 2
Placar de Tênis:
Jogador 1: 0 sets, 0 games, 30 pontos no game atual
Jogador 2: 0 sets, 0 games, 0 pontos no game atual
Próximo saque: Primeiro Jogador
Modo: Normal
Opções:
Pontuar Jogador 1: 1
Pontuar Jogador 2: 2
Placar de Tênis:
Jogador 1: 0 sets, 0 games, 40 pontos no game atual
Jogador 2: 0 sets, 0 games, 0 pontos no game atual
Próximo saque: Primeiro Jogador
Modo: Normal
Opções:
Pontuar Jogador 1: 1
Pontuar Jogador 2: 2
Placar de Tênis:
Jogador 1: 0 sets, 0 games, 40 pontos no game atual
Jogador 2: 0 sets, 0 games, 15 pontos no game atual
Próximo saque: Primeiro Jogador
Modo: Normal
Opções:
Pontuar Jogador 1: 1
Pontuar Jogador 2: 2
Placar de Tênis:
Jogador 1: 0 sets, 0 games, 40 pontos no game atual
Jogador 2: 0 sets, 0 games, 30 pontos no game atual
Próximo saque: Primeiro Jogador
Modo: Normal
Opções:
Pontuar Jogador 1: 1
Pontuar Jogador 2: 2
Placar de Tênis:
Jogador 1: 0 sets, 0 games, Deuce
Jogador 2: 0 sets, 0 games, Deuce
Próximo saque: Primeiro Jogador
Modo: Deuce
Opções:
Pontuar Jogador 1: 1
Pontuar Jogador 2: 2
